api/circulation.py

```diff
@@ -413,27 +415,32 @@ def fulfill(self, patron, pin, licensepool, delivery_mechanism, sync_on_failure=
         return fulfillment
 
     def fulfill_open_access(self, licensepool, delivery_mechanism):
-        # Keep track of a default way to fulfill this loan in case the
-        # patron's desired delivery mechanism isn't available.
+        """Fulfill an open-access LicensePool through the requested
+        DeliveryMechanism.
+
+        :param licensepool: The title to be fulfilled.
+        :param requested_lpdm: A DeliveryMechanism.
+        """
+        if isinstance(delivery_mechanism, LicensePoolDeliveryMechanism):
+            self.log.warn("LicensePoolDeliveryMechanism passed into fulfill_open_access, should be DeliveryMechanism.")
+            delivery_mechanism = delivery_mechanism.delivery_mechanism
```

Comment:
Here's a hack that fixes the problem if anyone else ever calls fulfill_open_access with a LPDM instead of a DM.